### PR TITLE
Updated metal-go client for sub-commands ports

### PR DESCRIPTION
**Pull request by codinja1188**
Breakout from https://github.com/equinix/metal-cli/pull/270

What this PR does / why we need it:

This PR replaces packngo with metal-go for all interactions with the Equinix Metal API specifically for ports sub commands

DISCUSSION POINTS:

Missing Includes and Excludes in Spec for the API "CreatePortVlanAssignmentBatch"

### CODE REVIEW

File: internal/ports/vlans.go [hunk 1]
```diff
@@ -50,28 +51,35 @@ func (c *Client) Vlans() *cobra.Command {
...
+				CreateInputVlanAssignmentsInner[0].SetNative(t)
+				req.SetVlanAssignments(CreateInputVlanAssignmentsInner)
 			}
-			if len(req.VLANAssignments) == 0 {
+
+			if len(req.GetVlanAssignments()) == 0 {
 				return errors.New("no VLAN assignments specified")
 			}
-			batch, _, err := c.VLANService.CreateBatch(portID, req, getOpts)
+			batch, _, err := c.PortService.CreatePortVlanAssignmentBatch(context.Background(), portID).PortVlanAssignmentBatchCreateInput(*req).I
```
Comment (ocobles) [inline internal/ports/vlans.go]:
```suggestion
			batch, _, err := c.PortService.CreatePortVlanAssignmentBatch(context.Background(), portID).PortVlanAssignmentBatchCreateInput(*req).Execute()
```

Comment (codinja1188) [reply internal/ports/vlans.go]: This needs to be addressed in spec side first missing Includes, Excludes

Comment (ctreatma) [reply internal/ports/vlans.go]: Any issues like this where you are aware that the `metal-go` client is missing features that are needed for conversion should be noted on the relevant PR and the PR should be placed in draft status.  That will protect against accidental merge of changes that are known to be not completely working, and will make it easier for the whole team to understand what issues are blocking the remaining `metal-go` conversion PRs.

**Comment by cprivitere** [comment]
Is this good to go or does it need to be set to draft?

File: internal/ports/convert.go [hunk 2]
```diff
@@ -53,26 +55,38 @@ func (c *Client) Convert() *cobra.Command {
 		// TODO: can we add ip-reservation-id?
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cmd.SilenceUsage = true
+			inc := []string{}
+			exc := []string{}
```
Comment (ctreatma) [inline internal/ports/convert.go:59]: These should be wired up to the `--include` and `--exclude` flags

File: internal/ports/retrieve.go [hunk 3]
```diff
@@ -41,17 +41,17 @@ func (c *Client) Retrieve() *cobra.Command {
 
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cmd.SilenceUsage = true
-			listOpts := c.Servicer.ListOptions(nil, nil)
+			inc := []string{}
+			exc := []string{}
```
Comment (ctreatma) [inline internal/ports/retrieve.go:45]: These should also be wired up to the `--include` and `--exclude` flags

**Comment by ctreatma** [comment]
> Missing Includes and Excludes in Spec for the API "CreatePortVlanAssignmentBatch"

This should be resolved as of `metal-go` `v0.16.0`.

**Comment by codinja1188** [comment]
Changes are handled in the following PR https://github.com/equinix/metal-cli/pull/355, close it

**Pull request closed without merging**